DietPi-LetsEncrypt | Initial Lighttpd-only support for multiple comma-separated domains

Ruben-VV opened the pull request:
Fixed problem when entering multiple domains separated by commas

<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
**Status**: Ready

**Reference**: https://github.com/MichaIng/DietPi/issues/1622 (only fixed problem with multiple domains)

**Commit list/description**:
<!--
- DietPi-Config | Add "Fan control" option to "Performance Options"
![Screenshot](https://xxx.github.com/images/xxx.png)
- DietPi-Config | Add "Fan control" support for Odroid C2
- DietPi-Config | Syntax fix
-->
- DietPi-LetsEncrypt | fixed problem with multiple domains separated by commas

Ruben-VV committed:
fix multiple domains separated by commas

Fixed problem when entering multiple domains separated by commas

MichaIng commented:
Many thanks. I guess you tested it with Lighttpd? Since with Apache and Nginx, using multiple domains should lead to invalid config files:
#### Apache:
```
G_CONFIG_INJECT 'ServerName[[:blank:]]' "ServerName $LETSENCRYPT_DOMAIN" "$fp_defaultsite" '<VirtualHost'
```
#### Nginx:
```
G_CONFIG_INJECT 'server_name[[:blank:]]' "	server_name $LETSENCRYPT_DOMAIN;" "$fp_defaultsite" 'listen[[:blank:]]'
```

For Apache, it would be thinkable to use the first entry as "main" servername and add subsequent entries as `ServerAlias`. Not sure which works (best) with the Certbot Apache2 module.

On Nginx multiple names can be added space separated to via `server_name` directive, so the commas must be replaced then: https://nginx.org/en/docs/http/ngx_http_core_module.html#server_name

I'll see if I find time to add this, should be tested then and the information added to the input box.

Ruben-VV committed:
fix previous commit 4cf5f42

Corrected the fix 4cf5f42. Only apply to Lighttpd.

Ruben-VV commented:
> 
> 
> Many thanks. I guess you tested it with Lighttpd? Since with Apache and Nginx, using multiple domains should lead to invalid config files:
> #### Apache:
> 
> ```
> G_CONFIG_INJECT 'ServerName[[:blank:]]' "ServerName $LETSENCRYPT_DOMAIN" "$fp_defaultsite" '<VirtualHost'
> ```
> 
> #### Nginx:
> 
> ```
> G_CONFIG_INJECT 'server_name[[:blank:]]' "	server_name $LETSENCRYPT_DOMAIN;" "$fp_defaultsite" 'listen[[:blank:]]'
> ```
> 
> For Apache, it would be thinkable to use the first entry as "main" servername and add subsequent entries as `ServerAlias`. Not sure which works (best) with the Certbot Apache2 module.
> 
> On Nginx multiple names can be added space separated to via `server_name` directive, so the commas must be replaced then: https://nginx.org/en/docs/http/ngx_http_core_module.html#server_name
> 
> I'll see if I find time to add this, should be tested then and the information added to the input box.

Yes, I tested it in Lighttpd and it work well.
I have made another change so that it only applies in Lighttpd

MichaIng committed:
v6.32

+ DietPi-LetsEncrypt: Add explanation about simple Lighttpd multi domain support, slight more explicit/clear string manipulation method

MichaIng commented:
Okay, many thanks, I'll merge it as it does not hurt at all and simply works 👍. Full support including Apache2 and Nginx likely follows next version then, probably I can even add it during beta phase as method is clear already.

I slightly adjusted the method: `${var%%,*}` more explicitly means to cut away the longest match from "`,`" till end (`${var%,*}` would be shortest match). Its only to make it more definit what is meant since practically you code does the same, replaces the longest match by default.
